src/Uno.UI/Microsoft/UI/Xaml/Controls/RadioButtons/RadioButtons.cs

```diff
@@ -316,6 +316,12 @@ private void OnRepeaterElementPrepared(ItemsRepeater sender, ItemsRepeaterElemen
 					{
 						Select(args.Index);
 					}
+
+					//If SelectedItem/SelectedIndex has already been set by the time the elements are loaded, ensure we sync the selection
+					if (args.Index == SelectedIndex)
```

Comment:
Since this is imported code you should prefix the comment with UNO WORKAROUND.

Do we understand why this is needed with Uno but not UWP/WinUI?

Reply:
It may be that it has been fixed in the original source

Reply:
Seems like its an Uno specific need similar to the syncing of the selected state for NavigationView. Has to do with the timing of when ItemsRepeater.OnElementPrepared is called (https://github.com/unoplatform/uno/issues/4689)